chatlas/_auto.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+
+import json
+import os
+from typing import Callable, Literal, Optional
+
+from ._anthropic import ChatAnthropic, ChatBedrockAnthropic
+from ._chat import Chat
+from ._github import ChatGithub
+from ._google import ChatGoogle, ChatVertex
+from ._groq import ChatGroq
+from ._ollama import ChatOllama
+from ._openai import ChatAzureOpenAI, ChatOpenAI
+from ._perplexity import ChatPerplexity
+from ._snowflake import ChatSnowflake
+from ._turn import Turn
+
+AutoProviders = Literal[
+    "anthropic",
+    "bedrock-anthropic",
+    "github",
+    "google",
+    "groq",
+    "ollama",
+    "openai",
+    "azure-openai",
+    "perplexity",
+    "snowflake",
+    "vertex",
+]
+
+_provider_chat_model_map: dict[AutoProviders, Callable[..., Chat]] = {
+    "anthropic": ChatAnthropic,
+    "bedrock-anthropic": ChatBedrockAnthropic,
+    "github": ChatGithub,
+    "google": ChatGoogle,
+    "groq": ChatGroq,
+    "ollama": ChatOllama,
+    "openai": ChatOpenAI,
+    "azure-openai": ChatAzureOpenAI,
+    "perplexity": ChatPerplexity,
+    "snowflake": ChatSnowflake,
+    "vertex": ChatVertex,
+}
+
+
+def ChatAuto(
+    system_prompt: Optional[str] = None,
+    turns: Optional[list[Turn]] = None,
+    *,
+    provider: Optional[AutoProviders] = None,
+    model: Optional[str] = None,
+    **kwargs,
+) -> Chat:
+    """
+    Create a Chat instance using a provider determined by environment variables.
+
+    Create a Chat instance based on the specified provider, with optional system
+    prompt and conversation turns. The provider and model be specified either
+    through the function parameters or via the `CHATLAS_CHAT_PROVIDER` or
+    `CHATLAS_CHAT_MODEL` environment variables. Additional configuration can be
+    provided through kwargs or the `CHATLAS_CHAT_ARGS` environment variable (as
+    JSON).
+
+    The design of `ChatAuto()` allows you to easily switch between different
+    chat providers by changing the environment variable without modifying your
+    code:
+
+    * `system_prompt` and `turns` are always used, regardless of how
+      `provider` or the additional options are set. These values define
+      key behavior of your chat.
+
+    * When provided to `ChatAuto()`, `provider` and `kwargs` serve as
+      default parameters that are used when the associated `CHATLAS_CHAT_`
+      environment variables are not set.
+
+    * When `CHATLAS_CHAT_PROVIDER` or `CHATLAS_CHAT_MODEL` are set, they are
+      used in place of `provider` and `model`.
```

Comment:
I think these two points could be phrased better and come as the first bullet point(s)